clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp

```diff
@@ -5,3 +5,9 @@ static_assert(__is_same(decltype([] constexpr -> int { }( )), int)); // expected
 
 consteval int g() { } // expected-warning {{non-void function does not return a value}}
 static_assert(__is_same(decltype([] consteval -> int { }( )), int)); // expected-warning {{non-void lambda does not return a value}}
+
+namespace GH116485 {
+int h() {
+    if consteval { }
+} // expected-warning {{non-void function does not return a value}}
```

Comment:
Other test cases to add are:
```
void foo(int x) {
    if consteval {
    }
    switch (x) {
    case 1:
      foo(12); // Should get a `-Wimplicit-fallthrough` warning here
    case 2:
      break;
    }
}
```
https://godbolt.org/z/rns3Yjofq

or testing whether the consteval if actually causes an early return:
```
constexpr bool foo()  {
    if !consteval { return true; }
}

bool b = foo();
constinit bool c = foo();
```

Reply:
@AaronBallman thanks for the feedback. I’ve added tests, and the first one works as expected. Could you check the last one to see if I need to add any extra handling for that case?